File: RubberCity.Services/Services/EventService.cs
using Domain.Models;
using RubberCity.Data.Interfaces;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace RubberCity.Services.Services
{
    public class EventService
    {
        private readonly IEventRepository<Event> _eventRepository;

        public EventService(IEventRepository<Event> eventRepository)
        {
            _eventRepository = eventRepository;
        }

        public async Task<IEnumerable<Event>> GetAllEvents()
        {
            return await _eventRepository.GetAll();
        }

        public async Task<Event> GetEventById(int id)
        {
            return await _eventRepository.GetById(id);
        }

        public async Task AddEvent(Event eventEntity)
        {
            await _eventRepository.Add(eventEntity);
        }

        public async Task UpdateEvent(Event eventEntity)
        {
            await _eventRepository.Update(eventEntity);
        }

        public async Task DeleteEvent(int id)
        {
            var eventEntity = await _eventRepository.GetById(id);
            if (eventEntity != null)
            {
            await _eventRepository.Delete(eventEntity);
        }
    }
}
}

File: WWWRubberCityFoundation/src/app/services/event.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Event } from '../models/event';
import { endpoints } from './endpoints';

@Injectable({ providedIn: 'root' })
export class EventService {
  constructor(private http: HttpClient) {}

  getEvents(): Observable<Event[]> {
    return this.http.get<Event[]>(endpoints.eventsController.getAll);
  }

  getEventById(id: number): Observable<Event> {
    return this.http.get<Event>(`${endpoints.eventsController.getByID}/${id}`);
  }

  addEvent(event: Event): Observable<Event> {
    return this.http.post<Event>(endpoints.eventsController.add, event);
  }

  updateEvent(event: Event): Observable<Event> {
    return this.http.put<Event>(`${endpoints.eventsController.update}/${event.id}`, event);
  }

  deleteEvent(id: number): Observable<void> {
    return this.http.delete<void>(`${endpoints.eventsController.delete}/${id}`);
  }
}
